File: hospital/AMarotoChivite-HospitalTest/AMarotoChivite-HospitalTest-Java/src/main/java/HospitalCollections/base/IHospitalRepository.java
package HospitalCollections.base;

import HospitalCollections.models.Patient;

import java.util.List;

public interface IHospitalRepository extends ICrudRepository<Patient, String> {

    boolean isFull();

    int getNumPacients();

    int[] getNumByType();

    List<Patient> getAllOrderByEnterDate();

    List<Patient> getAllOrderByName();

    List<Patient> getAllOrderByType();

}

File: hospital/AMarotoChivite-HospitalTest/AMarotoChivite-HospitalTest-Kotlin/src/main/kotlin/HospitalCollections/base/IHospitalRepository.kt
package HospitalCollections.base

import HospitalCollections.models.Patient

interface IHospitalRepository : IGeneralRepository<Patient?, String?> {
    fun isFull(): Boolean

    val numPacients: Int

    val numByType: IntArray

    val allOrderByEnterDate: List<Patient?>

    val allOrderByName: List<Patient?>

    val allOrderByType: List<Patient?>
}